bin/check-vault-status.rb

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env ruby
+#
+# check-vault-status
+#
+# DESCRIPTION:
+#   This plugins checks if vault is up & reachable. It then checks
+#   if the vault is unsealed
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: vault
+#
+require 'sensu-plugin/check/cli'
+require 'vault'
+
+class VaultStatus < Sensu::Plugin::Check::CLI
+  option :host,
+         description: 'vault host',
+         short: '-h HOST',
+         long:  '--host HOST',
+         default: '127.0.0.1'
+
+  option :port,
+         description: 'vault port',
+         short: '-p PORT',
+         long: '--port PORT',
+         default: '8200'
```

Comment:
is there any particular reason this is a string and not an integer? Just asking as its easier to validate that something is an integer vs trying to cast a string and validate that it can be recast as an integer. If that makes sense we should use something like a:
```
default: 8200,
proc: proc(&:to_i)
```

This will immediately throw a type error if they supply something that can't be represented in an integer which would mean its not a valid port number. This is a pretty common pattern we try to follow with plugins, here is an example: https://github.com/sensu-plugins/sensu-plugins-http/blob/4.1.0/bin/check-http.rb#L87

Reply:
Yes, it should be an int